monai/data/image_writer.py

```diff
@@ -41,7 +42,69 @@
     PILImage, _ = optional_import("PIL.Image")
 
 
-__all__ = ["ImageWriter", "ITKWriter", "NibabelWriter", "PILWriter", "logger"]
+__all__ = [
+    "ImageWriter",
+    "ITKWriter",
+    "NibabelWriter",
+    "PILWriter",
+    "SUPPORTED_WRITERS",
+    "register_writer",
+    "resolve_writer",
+    "logger",
+]
+
+SUPPORTED_WRITERS: Dict = {}
+
+
+def register_writer(ext_name, *im_writer):
+    """
+    Register ``ImageWriter``, so that writing a file with filename extension ``ext_name``
+    could be resolved to a tuple of potentially appropriate ``ImageWriter``.
+    The customised writers could be registered by:
+
+    .. code-block:: python
+
+        from monai.data import image_writer
+        # `MyWriter` must implement `ImageWriter` interface
+        image_writer.register_writer(".nii", MyWriter)
+
+    Args:
+        ext_name: the filename extension of the image.
+            As an indexing key, it will be converted to a lower case string.
+        im_writer: one or multiple ImageWriter classes with high priority ones first.
+    """
+    fmt = f"{ext_name}".lower()
+    existing = look_up_option(fmt, SUPPORTED_WRITERS, default=())
+    all_writers = im_writer + existing
+    SUPPORTED_WRITERS[fmt] = all_writers
+
+
+def resolve_writer(ext_name, error_if_not_found=True) -> Sequence:
```

Comment:
I feel the `resolve_writer` is confusing at first look, actually it doesn't "resolve" thing, maybe `get_writer` is more clear?

Thanks.